src/Redirection/Dynamic/RedirectionManager.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace AppBundle\Redirection\Dynamic;
+
+use AppBundle\Entity\Redirection;
+use Doctrine\ORM\EntityManagerInterface;
+use JMS\Serializer\Serializer;
+use Psr\Cache\CacheItemPoolInterface;
+
+class RedirectionManager
+{
+    private $cache;
+    private $entityManager;
+    private $serializer;
+
+    public function __construct(CacheItemPoolInterface $cache, EntityManagerInterface $entityManager, Serializer $serializer)
+    {
+        $this->cache = $cache;
+        $this->entityManager = $entityManager;
+        $this->serializer = $serializer;
+    }
+
+    public function refreshRedirectionCache(Redirection $redirection): void
+    {
+        $item = $this->cache
+            ->getItem(md5($redirection->getFrom()))
```

Comment:
I think we can remove the `md5`. IMO it's not needed. Any reason you added it ?

Reply:
False we need it because of special chars like /? and some others reserved